src/Andoromeda.Kyubey.Portal/Services/TokenFileInfoRepository.cs

```diff
@@ -0,0 +1,207 @@
+                if (cultureSuffix == TokenCultureFileSuffix.ZHCN)
+                    availuableFilenames = fileNames.Where(x => x.Contains(TokenCultureFileSuffix.ZHTW)).ToArray();
+                if (availuableFilenames.Count() > 0)
+                    return availuableFilenames;
+
+                //zh-tw no file, -->zh-cn
+                if (cultureSuffix == TokenCultureFileSuffix.ZHTW)
+                    availuableFilenames = fileNames.Where(x => x.Contains(TokenCultureFileSuffix.ZHCN)).ToArray();
+                if (availuableFilenames.Count() > 0)
+                    return availuableFilenames;
+            }
+            //en
+            availuableFilenames = fileNames.Where(x => x.Contains(TokenCultureFileSuffix.EN)).ToArray();
+            if (availuableFilenames.Count() > 0)
+                return availuableFilenames;
+
+            //default
+            availuableFilenames = fileNames.Where(x => !AvailuableCulcureFileSuffix.Any(c => x.Contains(c))).ToArray();
+            return availuableFilenames;
+        }
+
+        public string[] GetTokenIncubationBannerPaths(string tokenId, string cultureStr)
+        {
+            tokenId.RemoveDangerousChar();
+            var folderPath = Path.Combine(tokenFolderAbsolutePath, tokenId, "slides");
+            var files = FileHelper.GetAllFileNameFromFolder(folderPath, "*.png");
+            var availableFiles = GetAvailableFileNames(files, cultureStr);
+            var availablePaths = availableFiles.Select(x => Path.Combine(tokenFolderRelativePath, x)).ToArray();
+            return availablePaths;
+        }
+
+        public string GetTokenIncubationDescription(string tokenId, string cultureStr)
+        {
+            tokenId.RemoveDangerousChar();
+            var folderPath = Path.Combine(tokenFolderAbsolutePath, tokenId, "incubator");
+            var files = FileHelper.GetAllFileNameFromFolder(folderPath, "description*.txt");
+            var availableFiles = GetAvailableFileNames(files, cultureStr);
+            var availablePath = availableFiles.Select(x => Path.Combine(folderPath, x)).FirstOrDefault();
+            if (availablePath != null)
+                return FileHelper.ReadAllText(availablePath);
+            return null;
+        }
+
+        public string GetTokenIncubationDetail(string tokenId, string cultureStr)
+        {
+            tokenId.RemoveDangerousChar();
+            var folderPath = Path.Combine(tokenFolderAbsolutePath, tokenId, "incubator");
+            var files = FileHelper.GetAllFileNameFromFolder(folderPath, "detail*.md");
+            var availableFiles = GetAvailableFileNames(files, cultureStr);
+            var availablePath = availableFiles.Select(x => Path.Combine(folderPath, x)).FirstOrDefault();
+            if (availablePath != null)
+                return FileHelper.ReadAllText(availablePath);
+            return null;
+        }
+
+        //public string GetTokenDescription(string tokenId, string cultureStr)
```

Comment:
Please don't check in any commented codes.

Reply:
done